core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -466,6 +470,12 @@ public ManagedChannelImpl shutdown() {
     if (!shutdown.compareAndSet(false, true)) {
       return this;
     }
+    channelExecutor.executeLater(new Runnable() {
+      @Override
+      public void run() {
+        channelStateManager.gotoState(SHUTDOWN);
```

Comment:
`delayedTransport.shutdown()` can do lots of things; it seems strange to single-out adding anything to the `channelExecutor`. But if you add a comment pointing out the subtle usage of `drain()` (I'm unsure whether here or at the drain is better) I could live with it.